pkg/cloudprovider/plugins.go

```diff
@@ -31,10 +33,19 @@ import (
 // the parameter is nil.
 type Factory func(config io.Reader) (Interface, error)
 
-// All registered cloud providers.
+// KMSServiceFactory is a function that returns a cloudprovider.kms.Service.
+// The cloud parameter is a cloudprovider.Interface which can be used by the kms.Service.
+// The config parameter provides the unstructured configuration specific
+// to the KMS service provider.
+type KMSServiceFactory func(cloud Interface, config map[string]interface{}) (kms.Service, error)
```

Comment:
@lavalamp I modified the PR slightly. Now, `cloud.go` contains no new code. All the KMSServices here in cloudproviders will simply be plugins, and will return an object satisfying `kms.Service` (where `kms` package is in `k8s.io/apiserver`).
 The only code that can be a code debt is in this file (the plugin code), and I believe it should be pretty static since it is just initialization code. Adding transformers can be done freely from inside the `plugins` folder.

Everything cloudprovider dependent is inside `plugins/` now, including the cloudprovider specific configuration of KMS.